sklearn_numba_dpex/kmeans/kernels.py

```diff
@@ -0,0 +1,727 @@
+import math
+from functools import lru_cache
+
+import numba_dpex as dpex
+import dpctl
+
+
+@lru_cache
+def make_initialize_to_zeros_1dim_kernel(n_samples, work_group_size, dtype):
```

Comment:
Regarding those two points:
- yes, since the shape of the input is available with `.shape` from within a kernel, however I suspected defining shapes as constants might be better for performance (regarding loop unrolling when those values are used to define loops). This could be confirmed with some testing or more educated explanations from `numba_dpex` devs. On the other and, the advantage of writing kernels for arbitrary input tensors is that it can save compilation time.
- I don't know

The answers would be interesting, but as also discussed with @ogrisel regarding other TODOs for optimizing all those small auxilliary kernels, it should be addressed (if ever) in future PRs and I see the priority as secondary, because the execution time of those kernels is only a minor fraction of the total execution time of the kmeans (the fused fixed kernel being the main cost).

Reply:
Please write an inline TODO comment to explain this.